protocol/communities_events_utils_test.go

```diff
@@ -912,68 +913,39 @@ func testAcceptMemberRequestToJoin(base CommunityEventsTestsInterface, community
 		"alice did not receive community request to join response",
 	)
 	s.Require().NoError(err)
-}
-
-func testAcceptMemberRequestToJoinNotConfirmedByControlNode(base CommunityEventsTestsInterface, community *communities.Community, user *Messenger) {
-	// set up additional user that will send request to join
-	_, err := user.Start()
-
-	s := base.GetSuite()
-
-	s.Require().NoError(err)
-	defer user.Shutdown() // nolint: errcheck
-
-	advertiseCommunityTo(s, community, base.GetControlNode(), user)
-
-	// user sends request to join
-	requestToJoin := &requests.RequestToJoinCommunity{CommunityID: community.ID()}
-	response, err := user.RequestToJoinCommunity(requestToJoin)
-	s.Require().NoError(err)
-	s.Require().NotNil(response)
-	s.Require().Len(response.RequestsToJoinCommunity, 1)
-
-	sentRequest := response.RequestsToJoinCommunity[0]
 
-	// event sender receives request to join
-	response, err = WaitOnMessengerResponse(
+	// event sender receives update community
+	_, err = WaitOnMessengerResponse(
 		base.GetEventSender(),
-		func(r *MessengerResponse) bool { return len(r.RequestsToJoinCommunity) > 0 },
-		"event sender did not receive community request to join",
+		func(r *MessengerResponse) bool {
+			return len(r.Communities()) > 0 && r.Communities()[0].HasMember(&user.identity.PublicKey)
+		},
+		"event sender did not receive community with the new member",
 	)
 	s.Require().NoError(err)
-	s.Require().Len(response.RequestsToJoinCommunity, 1)
-
-	// event sender has not accepted request yet
-	eventSenderCommunity, err := base.GetEventSender().GetCommunityByID(community.ID())
-	s.Require().NoError(err)
-	s.Require().False(eventSenderCommunity.HasMember(&user.identity.PublicKey))
 
-	acceptRequestToJoin := &requests.AcceptRequestToJoinCommunity{ID: sentRequest.ID}
-	response, err = base.GetEventSender().AcceptRequestToJoinCommunity(acceptRequestToJoin)
-	s.Require().NoError(err)
-	s.Require().NotNil(response)
-	s.Require().Len(response.Communities(), 1)
-	// we don't expect `user` to be a member already, because `eventSender` merely
-	// forwards its accept decision to the control node
-	s.Require().False(response.Communities()[0].HasMember(&user.identity.PublicKey))
+	for i := 0; i <= 5; i++ {
```

Comment:
`WaitOnMessengerResponse` already uses a retry mechanism (`RetryWithBackOff`) internally. Is this for loop really necessary?